chipsec/helper/osx/helper.py

```diff
@@ -237,43 +279,93 @@ def native_set_EFI_variable(self, name, guid, data, datasize, attrs=None):
     #########################################################
 
     def read_io_port(self, io_port, size):
-        raise NotImplementedError()
+        in_buf = struct.pack(_io_msg_t_fmt, io_port, size, 0)
+        out_buf = self.ioctl(IOCTL_RDIO,in_buf)
+        try:
+            if 1 == size:
+                value = struct.unpack(_io_msg_t_fmt,out_buf)[2] & 0xff
+            elif 2 == size:
+                value = struct.unpack(_io_msg_t_fmt,out_buf)[2] & 0xffff
+            else:
+                value = struct.unpack(_io_msg_t_fmt,out_buf)[2] & 0xffffffff
+        except:
+            logger().error("DeviceIoControl did not return value of proper size %x (value = '%s')" %(size,out_buf))
+        return value
 
     def write_io_port(self, io_port, value, size):
-        raise NotImplementedError()
+        in_buf = struct.pack(_io_msg_t_fmt, io_port, size, value)
+        return self.ioctl(IOCTL_WRIO,in_buf)
 
     def read_cr(self, cpu_thread_id, cr_number):
-        raise NotImplementedError()
+        #self.set_affinity(cpu_thread_id)
```

Comment:
Could you add a warning log here so that the user is aware that thread selection is not implemented? (same for below)